metrics_collector/client_go_adaper/metrics.go

```diff
@@ -0,0 +1,98 @@
+package client_go_adaper
+
+import (
+	"net/url"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	clientmetrics "k8s.io/client-go/tools/metrics"
+
+	"github.com/longhorn/longhorn-manager/metrics_collector/registry"
+)
+
+// this file contains setup logic to initialize the myriad of places
+// that client-go registers metrics.  We copy the names and formats
+// from Kubernetes so that we match the core controllers.
+const (
+	LonghornName = "longhorn"
+)
+
+var (
+	// client metrics
+	clientGoRestClientSubsystem = "rest_client"
```

Comment:
Can we name it specifically like `k8s_rest_client`?

Reply:
I am not so sure. We use this one for both kubeClient and longhornClient https://github.com/longhorn/longhorn-manager/blob/73f1914b988daa121599f7ff2c4c9bbf59a40967/controller/controller_manager.go#L49-L54

Reply:
ah, then all good :slightly_smiling_face: